### PR TITLE
Mediator logging

### PR DESCRIPTION
**Pull request by li-guy**
- Mediator to LEA TCP connection +disconnections logged (needed when keepalives are not responded to or not supported) to assist administrators.
- Replace MediatorName with MediatorID as it is used.
- Appended Mediator to log messages.

### CODE REVIEW

File: src/mediator/mediator.c [hunk 1]
```diff
@@ -437,12 +437,12 @@ static void destroy_med_state(mediator_state_t *state) {
...
-			close(state->timerev->fd);
-		}
-		free(state->timerev);
-	}
+  if (state->timerev) {
+    if (state->timerev->fd != -1) {
+      close(state->timerev->fd);
+    }
+    free(state->timerev);
+  }
```
Comment (salcock) [inline src/mediator/mediator.c]: If you get a chance, can you undo some of these whitespace-only changes?

File: src/mediator/mediator.c [hunk 2]
```diff
@@ -2898,13 +2907,13 @@ static void *start_pcap_thread(void *params) {
...
     }
-    logger(LOG_INFO, "OpenLI mediator: exiting pcap thread.");
+    logger(LOG_INFO, "OpenLI Mediator: exiting pcap thread.");
     pthread_exit(NULL);
 }
 
 int main(int argc, char *argv[]) {
     char *configfile = NULL;
-    char *mediatorid = NULL;
+    int *mediatorid = NULL;
```
Comment (salcock) [inline src/mediator/mediator.c]: This should be an "int" not an "int *".

File: src/mediator/mediator.c [hunk 3]
```diff
@@ -2922,7 +2931,7 @@ int main(int argc, char *argv[]) {
             { NULL, 0, 0, 0},
         };
 
-        int c = getopt_long(argc, argv, "c:dm:p:h", long_options, &optind);
+        int c = getopt_long(argc, argv, "c:d:m:p:h", long_options, &optind);
```
Comment (salcock) [inline src/mediator/mediator.c]: Are you sure? A colon after the letter means that there must be a value after the argument, but "-d" would normally be used on its own, without a following value.

File: src/mediator/mediator.c (hunk 2 above)
Comment (salcock) [reply src/mediator/mediator.c]: This also means that you'll need to use `strtoul` when you assign it inside the getopt block, e.g.

    case 'm':
        mediatorid = strtoul(optarg, NULL, 0);
        break;

**Comment by li-guy** [comment]
Incorporated the comments, please review

**Comment by salcock** [comment]
Merged into develop. Thanks for your help!

**Pull request merged**